### PR TITLE
Status leds round2

### PR DESCRIPTION
**Pull request by gerth2**
Continuation of https://github.com/PhotonVision/photonvision/pull/802

Support RGB status LED to indicate:

Running/no-running
NT connected
At least one target visible

### CODE REVIEW

File: photon-core/src/main/java/org/photonvision/common/hardware/HardwareManager.java [hunk 1]
```diff
@@ -160,21 +165,60 @@ public boolean restartDevice() {
...
+        blinkCounter++;
+    }
+
+    private ArrayList<Boolean> pipelineTargets = new ArrayList<Boolean>();
+    private boolean ntConnected = false;
+    private boolean systemRunning = false;
+    int blinkCounter = 0;
+
+    public void setTargetsVisibleStatus(int pipelineIdx, boolean hasTargets) {
+        pipelineTargets.ensureCapacity(pipelineIdx);
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/common/hardware/HardwareManager.java:220]: Isn't cameraIdx more accurate? Also, a map might be a better data structure here

File: photon-core/src/main/java/org/photonvision/common/hardware/StatusLED.java [hunk 2]
```diff
@@ -45,4 +45,14 @@ public StatusLED(List<Integer> statusLedPins) {
             blueLED = new CustomGPIO(statusLedPins.get(2));
         }
     }
+
+    public void setRGB(boolean r, boolean g, boolean b) {
+        // System.out.printf("DEBUG setting LED's to %b %b %b\n", r, g, b);
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/common/hardware/StatusLED.java:50]: guh

File: photon-core/src/main/java/org/photonvision/common/hardware/HardwareManager.java (hunk 1 above)
Comment (gerth2) [reply photon-core/src/main/java/org/photonvision/common/hardware/HardwareManager.java:220]: I like how I had amnesia that maps existed last night. Yes, map 1000% better.

File: photon-core/src/main/java/org/photonvision/common/hardware/StatusLED.java (hunk 2 above)
Comment (gerth2) [reply photon-core/src/main/java/org/photonvision/common/hardware/StatusLED.java:50]: gotta debug. I"m gonna snag hardware from our warehouse today to confirm it's actually pumping out values.

**Pull request merged**